src/msg/async/frames_v2.cc

```diff
@@ -317,6 +324,13 @@ Tag FrameAssembler::disassemble_preamble(bufferlist& preamble_bl) {
     m_descs[i].logical_len = preamble->segments[i].length;
     m_descs[i].align = preamble->segments[i].alignment;
   }
+
+  m_flags = preamble->flags;
+
+  if (is_compressed() && !m_compression->rx) {
+      throw FrameError(fmt::format("Incoming frame is compressed but compression handler is missing"));
```

Comment:
Huh, do we envision a scenario this can happen? If not, then matybe an assertion would be a batter way?

Reply:
Currently - never.
In theory - can happen in case of an error in protocol implementation. At the moment, what will happen is that following the FrameError, the connection will be closed and the client/server will try to re-connect . 

An assert might be more appropriate here, since the msgr cannot recover from this and will go into a loop.